homeassistant/components/overkiz/switch.py

```diff
@@ -0,0 +1,137 @@
+
+    turn_on: Callable[[Callable[..., Awaitable[None]]], Awaitable[None]]
+    turn_off: Callable[[Callable[..., Awaitable[None]]], Awaitable[None]]
+    is_on: Callable[[Callable[[str], OverkizStateType]], bool]
+
+
+@dataclass
+class OverkizSwitchDescription(SwitchEntityDescription, OverkizSwitchDescriptionMixin):
+    """Class to describe an Overkiz switch."""
+
+
+SWITCH_DESCRIPTIONS: list[OverkizSwitchDescription] = [
+    OverkizSwitchDescription(
+        key=UIWidget.DOMESTIC_HOT_WATER_TANK,
+        turn_on=lambda execute_command: execute_command(
+            OverkizCommand.SET_FORCE_HEATING, OverkizCommandParam.ON
+        ),
+        turn_off=lambda execute_command: execute_command(
+            OverkizCommand.SET_FORCE_HEATING, OverkizCommandParam.OFF
+        ),
+        is_on=lambda select_state: (
+            select_state(OverkizState.IO_FORCE_HEATING) == OverkizCommandParam.ON
+        ),
+        icon="mdi:water-boiler",
+    ),
+    OverkizSwitchDescription(
+        key=UIClass.ON_OFF,
+        turn_on=lambda execute_command: execute_command(OverkizCommand.ON),
+        turn_off=lambda execute_command: execute_command(OverkizCommand.OFF),
+        is_on=lambda select_state: (
+            select_state(OverkizState.CORE_ON_OFF) == OverkizCommandParam.ON
+        ),
+        device_class=SwitchDeviceClass.OUTLET,
+    ),
+    OverkizSwitchDescription(
+        key=UIClass.SWIMMING_POOL,
+        turn_on=lambda execute_command: execute_command(OverkizCommand.ON),
+        turn_off=lambda execute_command: execute_command(OverkizCommand.OFF),
+        is_on=lambda select_state: (
+            select_state(OverkizState.CORE_ON_OFF) == OverkizCommandParam.ON
+        ),
+        icon="mdi:pool",
+    ),
+    OverkizSwitchDescription(
+        key=UIWidget.RTD_INDOOR_SIREN,
+        turn_on=lambda execute_command: execute_command(OverkizCommand.ON),
+        turn_off=lambda execute_command: execute_command(OverkizCommand.OFF),
+        is_on=lambda select_state: (
+            select_state(OverkizState.CORE_ON_OFF) == OverkizCommandParam.ON
+        ),
+        icon="mdi:bell",
+    ),
+    OverkizSwitchDescription(
+        key=UIWidget.RTD_OUTDOOR_SIREN,
+        turn_on=lambda execute_command: execute_command(OverkizCommand.ON),
+        turn_off=lambda execute_command: execute_command(OverkizCommand.OFF),
+        is_on=lambda select_state: (
+            select_state(OverkizState.CORE_ON_OFF) == OverkizCommandParam.ON
+        ),
+        icon="mdi:bell",
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Overkiz switch from a config entry."""
+    data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[OverkizSwitch] = []
+
+    supported_devices = {
+        description.key: description for description in SWITCH_DESCRIPTIONS
+    }
+
```

Comment:
This could be a constant since it never changes

Reply:
```suggestion
    SUPPORTED_DEVICES = {
        description.key: description for description in SWITCH_DESCRIPTIONS
    }
```

Should we do it like this? (and later on, refactor this in all platforms?)

Reply:
Looks good. Other platforms can come later